iceoryx_posh/test/moduletests/test_roudi_config_toml_file_provider.cpp

```diff
@@ -14,83 +14,213 @@
 // limitations under the License.
 //
 // SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_dust/cxx/std_string_support.hpp"
 #include "iceoryx_posh/roudi/cmd_line_args.hpp"
 #include "iceoryx_posh/roudi/roudi_config_toml_file_provider.hpp"
 
-#include "iceoryx/tests/posh/moduletests/test_input_path.hpp"
 #include "test.hpp"
 
+#if __cplusplus >= 201703L
+#include <filesystem>
+#endif
```

Comment:
Unfortunately it is not easy to get the path to the temp directory in a cross-platform way without C++17. To be able to test if a file can actually be opened, one test is added which run only on the C++17 targets like Windows, macOS and FreeBSD.

To make it work with C++14 `experimental/filesystem` must be used but then one also needs to add some libraries for linking and therefore I went with the C++17 only route.

Reply:
That's ok, we'll move to C++17 soon anyway. Not sure if already with `v3.0`.